fix: cp-13.5.0 improve token tabs perf and move polling to parent component

salimtb opened the pull request:
<!--
Please submit this PR as a draft initially.
Do not mark it as "Ready for review" until the template has been completely filled out, and PR status checks have passed at least once.
-->

## **Description**

The new tabbed list will unmount and remount the lists when switching through the tabs.
This is _mostly_ okay, however the tokens tab performs some polling setup on mount (which is expensive if you do this frequently!).

Instead we lifted the polling up above the tab list so it is not re-fired on every mount.

DEV NOTE - ideally we should lift this all the way into our assets polling provider. And avoid component level updates.
- If components do need to update this, we can provide a context or hook to modify the chains to poll.

<!--
Write a short description of the changes included in this pull request, also include relevant motivation and context. Have in mind the following questions:
1. What is the reason for the change?
2. What is the improvement/solution?
-->

[![Open in GitHub Codespaces](https://github.com/codespaces/badge.svg)](https://codespaces.new/MetaMask/metamask-extension/pull/36642?quickstart=1)

## **Changelog**

<!--
If this PR is not End-User-Facing and should not show up in the CHANGELOG, you can choose to either:
1. Write `CHANGELOG entry: null`
2. Label with `no-changelog`

If this PR is End-User-Facing, please write a short User-Facing description in the past tense like:
`CHANGELOG entry: Added a new tab for users to see their NFTs`
`CHANGELOG entry: Fixed a bug that was causing some NFTs to flicker`

(This helps the Release Engineer do their job more quickly and accurately)
-->

CHANGELOG entry: null

## **Related issues**

Fixes:

## **Manual testing steps**

Try switching between the activity tab and tokens tab on the home screen.

## **Screenshots/Recordings**

<!-- If applicable, add screenshots and/or recordings to visualize the before and after of your change. -->

### **Before**

<!-- [screenshots/recordings] -->

### **After**

https://www.loom.com/share/2c484c4c70ce43f380b78c04843147c0?sid=cf2d5a50-4992-4e0c-be4e-4dd5c88d31a0

<!-- [screenshots/recordings] -->

## **Pre-merge author checklist**

- [x] I've followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Extension Coding Standards](https://github.com/MetaMask/metamask-extension/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.


<!-- CURSOR_SUMMARY -->
---

> [!NOTE]
> Moves EVM token balance polling from TokenList to AccountOverviewTabs, updates NFT detection to use selected chain IDs, and adjusts NFT test mocks.
> 
> - **Frontend**:
>   - **AccountOverviewTabs (`ui/components/multichain/account-overview/account-overview-tabs.tsx`)**:
>     - Add `useTokenBalances` polling using `getChainIdsToPoll` (casts to `Hex[]`).
>     - Update NFT detection to dispatch `detectNfts(selectedChainIds)`.
>   - **TokenList (`ui/components/app/assets/token-list/token-list.tsx`)**:
>     - Remove EVM token balance polling and related `getChainIdsToPoll` usage.
> - **Tests**:
>   - **NFT mocks (`test/e2e/tests/tokens/nft/mocks.js`)**:
>     - Adjust users tokens query, dropping `chainIds` param.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit 9ddf3816c4f74ba8bd1cc2205eefd25512f9f6a1. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>
<!-- /CURSOR_SUMMARY -->

salimtb committed:
fix: improve token tabs perf and move polling to parent component

Prithpal-Sooriya committed:
Merge branch 'main' of github.com:MetaMask/metamask-extension into fix/improve-tokens-tab-click

Prithpal-Sooriya committed:
refactor: remove log

Prithpal-Sooriya committed:
refactor: lint fix

Prithpal-Sooriya committed:
fix: update selector to use selected chains instead of all chains

juanmigdr reviewed (dismissed):
Nice catch from cursor! LGTM!

Prithpal-Sooriya committed:
test: fix e2e test mocks

nft detection is only on selected chains not all chains now, so making mocks a little looser
